### PR TITLE
fix: remove pg_notify usage in favor of polling

### LINKED ISSUE #2087: Switch schema notifications from db to use polling

### PR DESCRIPTION
**Pull request by wesbillman**
Fixes #2087

### CODE REVIEW

File: backend/controller/dal/notify.go [hunk 1]
```diff
@@ -51,6 +53,88 @@ type event struct {
...
+}
+
+func (d *DAL) pollDeployments(ctx context.Context) {
+	logger := log.FromContext(ctx)
+	retry := backoff.Backoff{}
+
+	previousDeployments := make(map[string]deploymentState)
+
+	for {
+		delay := time.Millisecond * 500
```
Comment (wesbillman) [inline backend/controller/dal/notify.go:82]: Do we want this to be configurable maybe?

File: backend/controller/dal/notify.go [hunk 2]
```diff
@@ -51,6 +53,88 @@ type event struct {
...
+	schemaHash  []byte
+	minReplicas int
+}
+
+func deploymentStateFromDeployment(deployment Deployment) (deploymentState, error) {
+	hasher := sha256.New()
+	data := []byte(deployment.Schema.String())
+	if _, err := hasher.Write(data); err != nil {
+		return deploymentState{}, fmt.Errorf("failed to hash schema: %w", err)
+	}
```
Comment (wesbillman) [inline backend/controller/dal/notify.go:67]: This is the hashing I had to change to. It seemed like `sha256.fromBytes` just didn't compute a different has at all, but maybe I was using it wrong.

File: backend/controller/controller.go [hunk 3]
```diff
@@ -1575,16 +1576,19 @@ func (s *Service) watchModuleChanges(ctx context.Context, sendChange func(respon
...
-				if err != nil {
+
+				hasher := sha.New()
+				data := []byte(moduleSchema.String())
+				if _, err := hasher.Write(data); err != nil {
 					return err
 				}
+
 				newState := moduleStateEntry{
-					hash:        sha256.FromBytes(moduleSchemaBytes),
```
Comment (wesbillman) [inline backend/controller/controller.go:1583]: This seemed to compute the same hash every time even if the schema changed.

**Review by gak — APPROVED** [review]

File: backend/controller/dal/notify.go (hunk 1 above)
Comment (matt2e) [reply backend/controller/dal/notify.go:82]: I vote for getting it in as is and adding an option if there's a need

**Review by matt2e — APPROVED** [review]
LGTM!

**Pull request merged**